Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Lib/DriverScore.hs

```diff
@@ -136,6 +140,19 @@ eventPayloadHandler merchantOpCityId DST.OnRideCompletion {..} = do
           pure (fromMaybe 0.0 ride.fare, incrementBonusEarningsBy, incrementLateNightTripsCountBy, 10)
     -- Esq.runNoTransaction $ do
     DSQ.incrementTotalEarningsAndBonusEarnedAndLateNightTrip (cast driverId) incrementTotalEarningsBy (incrementBonusEarningsBy + overallPickupCharges) incrementLateNightTripsCountBy
+
+    -- for extra fare mitigation --
+    merchantConfig <- SCTC.findByMerchantOpCityId merchantOpCityId (Just (DriverId (cast driverId))) >>= fromMaybeM (TransporterConfigNotFound merchantOpCityId.getId)
+    booking <- B.runInReplica $ BQ.findById ride.bookingId >>= fromMaybeM (BookingNotFound ride.bookingId.getId)
+    let ibConfig = IBM.getIssueBreachConfig EXTRA_FARE_MITIGATION merchantConfig
+    let allowedSTiers = ibConfig <&> (.ibAllowedServiceTiers)
+    let isRideAllowedForCounting = maybe False (\allowedServiceTiers -> null allowedServiceTiers || booking.vehicleServiceTier `elem` allowedServiceTiers) allowedSTiers
+    when isRideAllowedForCounting $
+      whenJust ibConfig $ \config -> do
+        IBM.incrementCompletedBookingCounterForIssueBreach EXTRA_FARE_MITIGATION ride.driverId (toInteger config.ibCountWindowSizeInDays)
+    driverInfo <- QDI.findById driverId >>= fromMaybeM DriverInfoNotFound
+    driver <- B.runInReplica $ QPerson.findById driverId >>= fromMaybeM (PersonDoesNotExist driverId.getId)
```

Comment:
Driver table Query can be avoided if `merchantId`, `driverId`, `merchantOperatingCityId` is passed from Parent fucntion.